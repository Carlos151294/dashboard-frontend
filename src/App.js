import React from 'react';
import { Route, Switch, Redirect } from 'react-router-dom';
import { useSelector } from 'react-redux';

// Styles
import './App.scss';

// Pages
import DashboardPage from './pages/dashboard/dashboard.component';
import LoginPage from './pages/loginpage/loginpage.component';
import Header from './components/header/header.component';
import ContactPage from './pages/contactpage/contactpage.component';

function App() {
  const { isLogged } = useSelector(state => state.auth);

  return (
    <main className="app-container">
      <Header />
      <Switch>
        <Route exact path='/' render={() => isLogged ? (<Redirect to='dashboard' />) : (<Redirect to='login' />)} />
        <Route path='/login' render={() => isLogged ? (<Redirect to='dashboard' />) : (<LoginPage />)} />
        <Route path='/dashboard' render={() => isLogged ? (<DashboardPage />) : (<Redirect to='login' />)} />
        <Route path='/contact' component={ContactPage} />
        <Route render={() => <Redirect to='login' />} />
      </Switch>
    </main>
  );
}

export default App;
