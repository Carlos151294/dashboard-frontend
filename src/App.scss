@import './assets/custom_styles/variables';

html {
	height: 100%;
}

body {
	height: inherit;
	font-family: 'Open Sans Condensed', sans-serif;
	padding: 20px 60px;
	margin: 0;

	@media only screen and (max-width: $bp-small) {
		padding: 20px 10px;
	}
}

a {
	text-decoration: none;
	color: black;
}

* {
	box-sizing: border-box;
}

main.app-container {
	height: inherit;
}

div#root {
	height: inherit;
}
