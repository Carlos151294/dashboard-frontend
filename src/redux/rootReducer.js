import { combineReducers } from 'redux';
import authReducer from './auth/auth.reducer';
import publicationsReducer from './publications/publications.reducer';

const rootReducer = combineReducers({
    auth: authReducer,
    publications: publicationsReducer
});

export default rootReducer;