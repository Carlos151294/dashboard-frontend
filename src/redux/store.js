import { applyMiddleware, createStore, compose } from 'redux';
import rootReducer from './rootReducer';
import thunk from 'redux-thunk';

const enhancers = [];
const middlewares = [thunk];

if (process.env.NODE_ENV === 'development') {
    enhancers.push(window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__());
}

const store = createStore(rootReducer, compose(applyMiddleware(...middlewares), ...enhancers));

export { store };