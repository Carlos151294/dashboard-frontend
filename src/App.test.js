import React from 'react';
import { Provider } from 'react-redux';
import { shallow, mount } from 'enzyme';
import { MemoryRouter } from 'react-router';

import App from './App';
import { store } from './redux/store';
import DashboardPage from './pages/dashboard/dashboard.component';
import LoginPage from './pages/loginpage/loginpage.component';
import ContactPage from './pages/contactpage/contactpage.component';

it('Excepect to render App component without crashing', () => {
    expect(shallow(<Provider store={store}><App /></Provider>)).toMatchSnapshot();
});

it('Should redirect to Login Page', () => {
    const wrapper = mount(
        <Provider store={store}>
            <MemoryRouter initialEntries={['/login']}>
                <App />
            </MemoryRouter>
        </Provider>
    );
    expect(wrapper.find(LoginPage)).toHaveLength(1);
});

it('Should redirect to Contact Page', () => {
    const wrapper = mount(
        <Provider store={store}>
            <MemoryRouter initialEntries={['/contact']}>
                <App />
            </MemoryRouter>
        </Provider>
    );
    expect(wrapper.find(ContactPage)).toHaveLength(1);
});

it('Should redirect to Login Page when user is NOT logged', () => {
    const wrapper = mount(
        <Provider store={store}>
            <MemoryRouter initialEntries={['/dashboard']}>
                <App />
            </MemoryRouter>
        </Provider>
    );
    expect(wrapper.find(DashboardPage)).toHaveLength(0);
});

it('Should redirect to Login Page for nonexitant route', () => {
    const wrapper = mount(
        <Provider store={store}>
            <MemoryRouter initialEntries={['/random']}>
                <App />
            </MemoryRouter>
        </Provider>
    );
    expect(wrapper.find(LoginPage)).toHaveLength(1);
});