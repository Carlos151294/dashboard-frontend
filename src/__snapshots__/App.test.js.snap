// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Excepect to render App component 1`] = `ShallowWrapper {}`;

exports[`Excepect to render App component without crashing 1`] = `ShallowWrapper {}`;
